packages/react-error-overlay/src/utils/warnings.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+import type { ReactFrame } from '../effects/proxyConsole';
+
+// This is a list of information to remove from React warnings, it's not particularly useful to show.
+const removals = [/Check your code at (.*?:.*)[.]/];
+
+function massage(
+  warning: string,
+  frames: ReactFrame[] | void
+): { message: string, stack: string } | null {
+  if (!frames) {
+    return null;
+  }
+
+  let message = warning;
+  const nIndex = message.indexOf('\n');
```

Comment:
This will crash if warning is not a string. Technically user can call `warning` with any types—we need to be more careful.